src/rlimitfloatenv.cpp

```diff
@@ -0,0 +1,82 @@
+#include <fenv.h>
+#include <sys/time.h>
+#include <sys/resource.h>
+#include "shareddata.h"
+#include "config.h"
+#include "dmtcp.h"
+
+/*************************************************************************
+ *
+ *  Save and restore rlimit and float exception settings.
+ *
+ *************************************************************************/
+
+namespace dmtcp
+{
+
+static int roundingMode = -1;
+static fenv_t envp;
+static rlim_t rlim_cur_nofile = 0;
+static rlim_t rlim_cur_stack = 0;
+
+static void
+save_rlimit_float_settings()
+{ roundingMode = fegetround();
+  fegetenv(&envp);
+  { struct rlimit rlim = {0, 0};
```

Comment:
Not sure if the braces here are intentional. If they are, please insert a newline after the brace.